### PR TITLE
feat(SBOMER-271): CycloneDX 1.6: SBOMer can't not parse .dependencies…

### PR DESCRIPTION
**Pull request by vibe13**
…[].provides

### CODE REVIEW

**Review by goldmann — APPROVED** [review]

File: core/src/main/java/org/cyclonedx/model/Dependency.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+public class Dependency extends BomReference {
+
+    @JsonProperty("dependsOn")
+    @JacksonXmlElementWrapper(useWrapping = false)
+    @JacksonXmlProperty(localName = "dependency")
+    private List<Dependency> dependencies;
+
+    @JsonProperty("provides")
+    @JacksonXmlElementWrapper(useWrapping = false)
+    @JacksonXmlProperty(localName = "dependency")
```
Comment (goldmann) [inline core/src/main/java/org/cyclonedx/model/Dependency.java:61]: I'm not sure if this is correct.

Comment (goldmann) [reply core/src/main/java/org/cyclonedx/model/Dependency.java:61]: https://cyclonedx.org/docs/1.6/xml/#type_dependencyType

Comment (vibe13) [reply core/src/main/java/org/cyclonedx/model/Dependency.java:61]: I have tried to get it from the spec https://github.com/CycloneDX/cyclonedx-core-java/blob/master/src/main/resources/bom-1.6.schema.json#L1847 and it looks like `dependsOn` and `provides` have the same definition...but did not find any pointers for the xml schema.

Comment (goldmann) [reply core/src/main/java/org/cyclonedx/model/Dependency.java:61]: Yah, for our use case it doesn't matter, but when submitting upstream, proper XML support may be necessary. We can always ask them to help us understand how it should be done for XML.

**Pull request merged**